eclair-core/src/main/scala/fr/acinq/eclair/payment/Bolt12Invoice.scala

```diff
@@ -103,65 +80,54 @@ case class PaymentBlindedRoute(route: Sphinx.RouteBlinding.BlindedRoute, payment
 
 object Bolt12Invoice {
   val hrp = "lni"
+  val signatureTag: ByteVector = ByteVector(("lightning" + "invoice" + "signature").getBytes)
   val DEFAULT_EXPIRY_SECONDS: Long = 7200
-  val DEFAULT_MIN_FINAL_EXPIRY_DELTA: CltvExpiryDelta = CltvExpiryDelta(18)
 
   /**
    * Creates an invoice for a given offer and invoice request.
    *
-   * @param offer    the offer this invoice corresponds to
    * @param request  the request this invoice responds to
    * @param preimage the preimage to use for the payment
    * @param nodeKey  the key that was used to generate the offer, may be different from our public nodeId if we're hiding behind a blinded route
    * @param features invoice features
    * @param paths    the blinded paths to use to pay the invoice
    */
-  def apply(offer: Offer,
-            request: InvoiceRequest,
+  def apply(request: InvoiceRequest,
             preimage: ByteVector32,
             nodeKey: PrivateKey,
-            minFinalCltvExpiryDelta: CltvExpiryDelta,
             features: Features[InvoiceFeature],
             paths: Seq[PaymentBlindedRoute]): Bolt12Invoice = {
-    require(request.amount.nonEmpty || offer.amount.nonEmpty)
-    val amount = request.amount.orElse(offer.amount.map(_ * request.quantity)).get
-    val tlvs: Seq[InvoiceTlv] = Seq(
-      Some(Chain(request.chain)),
-      Some(OfferId(offer.offerId)),
-      Some(Amount(amount)),
-      Some(Description(offer.description)),
-      if (!features.isEmpty) Some(FeaturesTlv(features.unscoped())) else None,
-      Some(Paths(paths.map(_.route))),
-      Some(PaymentPathsInfo(paths.map(_.paymentInfo))),
-      offer.issuer.map(Issuer),
-      Some(NodeId(nodeKey.publicKey)),
-      request.quantity_opt.map(Quantity),
-      Some(PayerKey(request.payerKey)),
-      request.payerNote.map(PayerNote),
-      Some(CreatedAt(TimestampSecond.now())),
-      Some(PaymentHash(Crypto.sha256(preimage))),
-      Some(Cltv(minFinalCltvExpiryDelta)),
-      request.payerInfo.map(PayerInfo),
-      request.replaceInvoice.map(ReplaceInvoice),
+    require(request.amount.nonEmpty || request.offer.amount.nonEmpty)
+    val amount = request.amount.orElse(request.offer.amount.map(_ * request.quantity)).get
+    val tlvs: Seq[InvoiceTlv] = removeSignature(request.records).records.toSeq ++ Seq(
+      Some(InvoicePaths(paths.map(_.route))),
+      Some(InvoiceBlindedPay(paths.map(_.paymentInfo))),
+      Some(InvoiceCreatedAt(TimestampSecond.now())),
+      Some(InvoiceRelativeExpiry(300)),
```

Comment:
We shouldn't have a hard-coded value here, that should be provided in the constructor (we have a configurable value for that in `NodeParams`: `invoiceExpiry`).